itkachuk/HomeTask6/RequireJSScaffold/scripts/users.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Created by acolodnitchii on 2/13/2015.
+ */
+define(["jquery","underscore","userService"],function($,_, userService){
+
+    var usersArray = {};
+
+    var updateUsersList = function updateUsersList(){
+        userService.getAll().done(function(users){
+            var templateContent = $("#usersTemplate").html();
+            var evaluated = _.template(templateContent)({users: users});
+
+            $("#usersList").html(evaluated);
+            usersArray = users;
+        });
+    };
+
+    var clearUserForm = function clearUserForm(){
+        $("#firstName").val("");
+        $("#lastName").val("");
+        $("#login").val("");
+        $("#active").prop("checked", false);
+        $("#addUserForm").prop("userId", null);
+        $("#submitButton").html("Submit");
+    };
+
+    return {
+        init: function(){
+            $("#addUser").bind("click",function(){
+                var addUserForm = $("#addUserForm");
+                addUserForm.toggle();
+                addUserForm.prop("userId", null);
+                $("#submitButton").html("Submit");
+            });
+
+            $("#cancelButton").bind("click",function(){
+                clearUserForm();
+            });
+
+            $("#addUserForm").submit(function(e){
+                var user = {
+                    firstName: $("#firstName").val(),
+                    lastName: $("#lastName").val(),
+                    login: $("#login").val(),
+                    active: $("#active").prop("checked")
+                };
+                if ($(this).prop("userId")) {
+                    user.id = $(this).prop("userId");
+                    userService.update(user)
+                        .done(function () {
+                            alert("user was updated");
+                            clearUserForm();
+                            updateUsersList();
+                            $("#addUserForm").hide();
+                        })
+                        .fail(function () {
+                            alert("error: user wasn't updated");
+                        });
+                } else {
+                    userService.add(user)
+                        .done(function () {
+                            alert("user was added");
+                            clearUserForm();
+                            updateUsersList();
+                            $("#addUserForm").hide();
+                        })
+                        .fail(function () {
+                            alert("error: user wasn't added");
+                        });
+                }
+                e.preventDefault();
+            });
+
+            $("#usersList").click(function(e){
```

Comment:
it is better to trigger specific functions for the buttons instead of analyzing e.target.tagName && e.target.name